docarray/doc_index/abstract_doc_index.py

```diff
@@ -693,3 +703,35 @@ def _to_numpy(self, val: Any) -> Any:
             return val.numpy()
         else:
             raise ValueError(f'Unsupported input type for {type(self)}: {type(val)}')
+
+    def _convert_dict_to_doc(
```

Comment:
I wonder if pydantic already offers something liket this out of the box.
And if not, maybe this should be a global and public helper function that we offer?